yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -849,24 +849,30 @@ def fixup_webp(self, info, idx=-1):
                     info['__files_to_move'].pop(thumbnail_filename), 'webp')
 
     def convert_thumbnail(self, thumbnail_filename, ext):
-        if ext != 'jpg':
-            raise FFmpegPostProcessorError('Only conversion to jpg is currently supported')
+        if ext == 'jpg':
+            format_name = 'JPEG'
+            opts = ['-bsf:v', 'mjpeg2jpeg']
+        elif ext == 'png':
+            format_name = 'PNG'
+            opts = []
+        else:
+            raise FFmpegPostProcessorError('Only conversion to either jpg or png is currently supported')
```

Comment:
We could create properties in `FFmpegThumbnailsConvertorPP` for the supported extensions and options:
```py
    _SUPPORTED_FORMATS = {
        'png': [],
        'jpg': ['-bsf:v', 'mjpeg2jpeg'],
    }
    
    SUPPORTED_EXTS = _SUPPORTED_FORMATS.keys()
```

Then, this code can be just:

```
opts = self._SUPPORTED_FORMATS.get(ext)
if opts is None:
    raise FFmpegPostProcessorError('Only conversion to %s is currently supported' % '/'.join(self.SUPPORTED_EXTS)
```

We can also replace all the other `in ('jpg', 'png')` with `in FFmpegThumbnailsConvertorPP.SUPPORTED_EXTS`. Even the `options` help string can use this. This will make it easier to expand in the future

Reply:
That's exactly what I was thinking! Although, I wanted to keep the changes small, so I tried to follow what was already implemented and tried not to stray too much.

But, honestly, yes, I believe that we should move the supported formats over to a separate property that we can then reference elsewhere. We should probably consider doing that for most of the other hardcoded "supported formats" in other options like for `opts.extractaudio` and `opts.convertsubtitles`, but that's getting off-topic.

Reply:
> We should probably consider doing that for most of the other hardcoded "supported formats" in other options like for opts.extractaudio and opts.convertsubtitles

Now that I look at the classes, there is no checks for the format in the postprocessor itself other than a few special cases. The PP are depending on `yt_dlp` do any necessary validation. Maybe I'll do the same for this. Currently there are 3 separate checks `ext in ('png', 'jpg')`

---

anyway, back to the point. I'll be creating the properties anyway. But I guess it doesn't have to part of this PR to keep changes minimal

Reply:
https://github.com/yt-dlp/yt-dlp/commit/81a23040ebf330a87a7eb842aa76884d5fc6e504